openedx/adg/lms/applications/admin.py

```diff
@@ -68,7 +105,403 @@ class BusinessLineAdmin(admin.ModelAdmin):
+        elif application.resume:
+            fieldset_title = RESUME_ONLY
+            resume_cover_letter_file_fields.append(RESUME)
+        else:
+            fieldset_title = COVER_LETTER_ONLY
+            if application.cover_letter_file:
+                resume_cover_letter_file_fields.append(COVER_LETTER_FILE)
+
+        resume_cover_letter_display_fields = self._get_resume_cover_letter_display_fields(application)
+
+        resume_cover_letter_fields = resume_cover_letter_file_fields + resume_cover_letter_display_fields
+        fieldset = fieldset_title, {'fields': tuple(resume_cover_letter_fields)}
+
+        return fieldset
+
+    def _get_resume_cover_letter_display_fields(self, application):
+        """
+        Get display fields, if applicable, for resume and/or cover letter
+
+        Arguments:
+            application (UserApplication): Application under review
+
+        Returns:
+            list: Display fields
+        """
+        resume_cover_letter_display_fields = []
+
+        if application.resume:
+            if is_displayable_on_browser(application.resume):
+                resume_cover_letter_display_fields.append(RESUME_DISPLAY)
+
+        if application.cover_letter_file:
+            if is_displayable_on_browser(application.cover_letter_file):
+                resume_cover_letter_display_fields.append(COVER_LETTER_FILE_DISPLAY)
+        elif application.cover_letter:
+            resume_cover_letter_display_fields.append(COVER_LETTER_TEXT)
+
+        return resume_cover_letter_display_fields
+
+    readonly_fields = (
+        EMAIL,
+        LOCATION,
+        LINKED_IN_PROFILE,
+        IS_SAUDI_NATIONAL,
+        GENDER,
+        PHONE_NUMBER,
+        DATE_OF_BIRTH,
+        ORGANIZATION,
+        APPLYING_TO,
+        RESUME,
+        COVER_LETTER_FILE,
+        COVER_LETTER_FILE_DISPLAY,
+        RESUME_DISPLAY,
+        COVER_LETTER_TEXT,
+        PREREQUISITES
+    )
+
+    def get_formsets_with_inlines(self, request, obj=None):
+        """
+        Override method `get_formsets_with_inlines` of ModelAdmin.
+
+        Override is needed to ensure that if the applicant has attached a resume, the education and work experience
+        inlines should not be rendered. If resume is not attached with application, EducationInline should be rendered
+        while WorkExperienceInline should be optionally rendered depending on whether the user has entered any.
+
+        Arguments:
+            request (WSGIRequest): HTTP request accessing application review page
+            obj (UserApplication): Application under review
+
+        Returns:
+            Formsets with inlines
+        """
+        if obj.resume:
+            return
+        for inline in self.get_inline_instances(request, obj):
+            if isinstance(inline, WorkExperienceInline)\
+                    and not WorkExperience.objects.filter(user_application=obj).exists():
```

Comment:
Can we use parentheses to break this line as preferred by pep8?

Reply:
I feel that looks more confusing:

```
            if isinstance(
                inline, WorkExperienceInline
            ) and not WorkExperience.objects.filter(user_application=obj).exists():
                continue
            yield inline.get_formset(request, obj), inline
```
OR
```
            if isinstance(inline, WorkExperienceInline) and not WorkExperience.objects.filter(
                user_application=obj
            ).exists():
                continue
            yield inline.get_formset(request, obj), inline
```

Would have been more appropriate if there was a single lengthy statement. Now there are two different statements separated by `and`, so I feel breaking before the operator is more readable. Plus it doesn't violate any PEP8 rules either, as suggested by the pycodestyle report.

Reply:
```suggestion
            if (isinstance(inline, WorkExperienceInline)
            and not WorkExperience.objects.filter(user_application=obj).exists()):
```

Actually pep8 prefers to wrap the long statement instead of using back slashes.

_The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation._

Reply:
The suggestion cannot be adopted as it results in an indentation error. Yes, pep8 discourages the use of backslashes but also follows with:

"Backslashes may still be appropriate at times."

In any case, I have created a new property and moved this logic there. Now this is how the code looks like here:

```
        for inline in self.get_inline_instances(request, obj):
            if isinstance(inline, WorkExperienceInline) and obj.has_no_work_experience:
                continue

            yield inline.get_formset(request, obj), inline
```

^This is a lot cleaner. Thanks for pointing it out!